src/DynamoCore/Search/NodeSearchModel.cs

```diff
@@ -233,8 +233,8 @@ internal string ProcessNodeCategory(string category, ref SearchElementGroup grou
 
         internal IEnumerable<NodeSearchElement> Search(string search, LuceneSearchUtility luceneSearchUtility)
         {
-            
-            if (luceneSearchUtility != null)
+            if (luceneSearchUtility == null) return null;
+            lock (luceneSearchUtility)
```

Comment:
is this still required in addition to the searches operations taking a lock out on the queue token?

Reply:
The lock is in case Search is used in parallel. As of now that could only happen if the user searches in the library and workspace simultaneously, since both have independent serial queues. We could mandate a unified search queue (for library, inCanvasSearch, and  in the future) or have this lock to allow several search queues in parallel.